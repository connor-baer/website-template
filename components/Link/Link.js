import React from 'react';
import NextLink from 'next/link';
import { useRouter } from 'next/router';
import url from 'url';

function withPreview(urlObj = {}, preview) {
  const { query = {} } = urlObj;
  return preview ? { ...urlObj, query: { ...query, preview } } : urlObj;
}

export default function Link(props) {
  const { query = {} } = useRouter();
  const hrefObj = url.parse(props.href);
  const asObj = url.parse(props.as || props.href);
  const hrefWithPreview = withPreview(hrefObj, query.preview);
  const asWithPreview = withPreview(asObj, query.preview);

  return (
    <NextLink {...props} href={hrefWithPreview} as={asWithPreview} passHref />
  );
}
