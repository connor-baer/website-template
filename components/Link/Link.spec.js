import React from 'react';
import * as Router from 'next/router';

import Link from './Link';

describe('Link', () => {
  function renderLink(fn, props = {}) {
    return fn(<Link {...props} />);
  }

  it('should render with default styles', () => {
    const href = 'https://example.com';
    const children = 'example';
    const { container } = renderLink(render, { href, children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with preview', () => {
    Router.useRouter = () => ({ query: { preview: true } });
    const href = 'https://example.com';
    const children = 'example';
    const { container } = renderLink(render, { href, children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const href = 'https://example.com';
    const children = 'example';
    const html = renderLink(renderToHtml, { href, children });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
