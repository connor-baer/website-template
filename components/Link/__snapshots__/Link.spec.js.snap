// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render with default styles 1`] = `
<a
  href="https://example.com/"
>
  example
</a>
`;

exports[`Link should render with preview 1`] = `
<a
  href="https://example.com/?preview=true"
>
  example
</a>
`;
